Modules/_functoolsmodule.c

```diff
@@ -115,25 +195,71 @@ partial_new(PyTypeObject *type, PyObject *args, PyObject *kw)
 
     pto->fn = Py_NewRef(func);
 
-    nargs = PyTuple_GetSlice(args, 1, PY_SSIZE_T_MAX);
-    if (nargs == NULL) {
+    pto->placeholder = state->placeholder;
+    if (new_nargs && PyTuple_GET_ITEM(args, new_nargs) == pto->placeholder) {
```

Comment:
I would defer the `new_nargs--` to here because it's closer to it's usage. Or I would just use `new_nargs > 1` and `new_nargs - 1`